tools/run_tests/run_xds_tests.py

```diff
@@ -950,6 +969,289 @@ def prepare_services_for_urlmap_tests(gcp, original_backend_service,
+        # test MATCH_ALL: super set labels won't match
+        [
+          {
+            'priority': 0,
+            'matchRules': [
+                {
+                    'prefixMatch': '/',
+                    'metadataFilters': [
+                        {
+                            'filterMatchCriteria': 'MATCH_ALL',
+                            'filterLabels': not_match_labels + match_labels
+                        }
+                    ]
+                }
+            ],
+            'service': original_backend_service.url
+          },
+          {
+            'priority': 1,
+            'matchRules': [
+                {
+                    'prefixMatch': '/',
+                    'metadataFilters': [
+                        {
+                            'filterMatchCriteria': 'MATCH_ANY',
+                            'filterLabels': not_match_labels + match_labels
+                        }
+                    ]
+                }
+            ],
+            'service': alternate_backend_service.url
+          },
+        ],
+        # test MATCH_ANY
+        [
+          {
+            'priority': 0,
+            'matchRules': [
+                {
+                    'prefixMatch': '/',
+                    'metadataFilters': [
+                        {
+                            'filterMatchCriteria': 'MATCH_ANY',
+                            'filterLabels': not_match_labels
+                        }
+                    ]
+                }
+            ],
+            'service': original_backend_service.url
+          },
+          {
+            'priority': 1,
+            'matchRules': [
+                {
+                    'prefixMatch': '/',
+                    'metadataFilters': [
+                        {
+                            'filterMatchCriteria': 'MATCH_ANY',
+                            'filterLabels': not_match_labels + match_labels
+                        }
+                    ]
+                }
+            ],
+            'service': alternate_backend_service.url
+          },
+        ]
+        ]
+
+        for route_rules in test_route_rules:
+            wait_until_all_rpcs_go_to_given_backends(original_backend_instances,
+                                                     _WAIT_FOR_STATS_SEC)
+            patch_url_map_backend_service(gcp, original_backend_service,
+                                          route_rules=route_rules)
+            wait_until_no_rpcs_go_to_given_backends(original_backend_instances,
+                                                    _WAIT_FOR_STATS_SEC)
+            wait_until_all_rpcs_go_to_given_backends(alternate_backend_instances,
+                                                     _WAIT_FOR_STATS_SEC)
+            patch_url_map_backend_service(gcp, original_backend_service)
+    finally:
+        patch_backend_service(gcp, alternate_backend_service, [])
+
+
+def test_api_listener(gcp, backend_service, instance_group,
```

Comment:
Can you name this func appropriately since it doesn't reflect the intent "LDS switches to the second API listener when the first one is deleted". Something like "test_duplicate_api_listener" or even better "test_duplicate_routing_config" (since user creates the  map+tp+fr pipeline and not the api listener)

Reply:
This test case is sort of mixed of corner cases of api listener. Not only duplicate api listener, but delete api listener case (L1151).

Reply:
ok